protoc-gen-pbandk/lib/src/commonMain/kotlin/pbandk/gen/Namer.kt

```diff
@@ -52,11 +52,14 @@ public interface Namer {
 
         override fun newEnumValueTypeName(enumTypeName: String, preferred: String, nameSet: Collection<String>): String {
             val typePrefix = splitWordsToSnakeCase(enumTypeName) + '_'
-            var name = splitWordsToSnakeCase(preferred).substringAfter(typePrefix)
-            name = name.uppercase()
+            var name = splitWordsToSnakeCase(preferred).let { snakeCase ->
+                snakeCase.substringAfter(typePrefix).takeUnless {
```

Comment:
This is a style nit, but I find the nested `let` and `takeUnless` calls here harder to read than if this was written imperatively:

```kt
var name = splitWordsToSnakeCase(preferred)
val nameWithoutPrefix = name.substringAfter(typePrefix)
if (!"^\\d|\\W".toRegex().matches(nameWithoutPrefix)) {
    name = nameWithoutPrefix
}
name = name.uppercase()
```